src/osha/oira/ploneintranet/quaive_links.py

```diff
@@ -0,0 +1,59 @@
+from Acquisition import aq_base
+from Products.Five import BrowserView
+
+import json
+import re
+
+
+class SurveyLinks(BrowserView):
+    attributes_checked = [
+        "description",
+        "introduction",
+        "solution_direction",
+        "legal_reference",
+        "action",
+        "action_plan",
+        "prevention_plan",
+        "requirements",
+    ]
+    url_regex = re.compile(
+        r"https?://[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b"
+        r"(?:[-a-zA-Z0-9()@:%_+.~#?&/=]*)"
+    )
+
+    def extract_links(self, obj):
+        """For the survey and each subobject in its content tree, list external
+        hyperlinks if it has them. Take care to return a datastructure with a
+        deterministic sort order.
+        """
+        links = []
+        for attrib in self.attributes_checked:
+            value = getattr(aq_base(obj), attrib, "")
+            if value:
+                links.extend(self.url_regex.findall(value))
+        # enforce deterministic sort order of links
+        links.sort()
+        if links:
+            yield {
+                "title": obj.Title(),
+                "url": obj.absolute_url(),
+                "path": "/".join(
+                    obj.getPhysicalPath()[len(self.context.getPhysicalPath()) :]
+                ),
+                # store the url in a nested dicationary, because we will later
+                # want to augment that dictionary with status information
+                "links": [{"url": url} for url in links],
+            }
+        if hasattr(obj, "objectIds"):
+            # reimplement objectValues() with deterministic sort order of sections
+            for child_id in sorted(obj.objectIds()):
```

Comment:
Why is it important to sort by id?

Reply:
Because the consumer will write extra information to the retrieved datastructure, and for that the data structure needs to provide consistent sorting. While that is cached on the consumer end, I'm rather safe than sorry and want to enforce sorting stability already on the endpoint.

Reply:
Actually I will solve this consumer-side, and streamline here.

Reply:
Actually I will keep the sorting here. It's more difficult to reconstruct that consumer side.

Reply:
Ok, but sorting by id is generally not a good sorting method. 
Anyway, I will refrain myself from reviewing this further.

Reply:
Oh? I did not know that. Just out of curiosity: what is the downside of sorting by id, and what would be better alternatives?